galsim/scene.py

```diff
@@ -0,0 +1,412 @@
+        >>>>     im_real.write('im_real_'+str(ind)+'.fits')
+        >>>>     im_param.write('im_param_'+str(ind)+'.fits')
+
+    This code snippet will draw images of the first 10 objects in the COSMOS catalog, at slightly
+    lower resolution than in COSMOS, with a real image and its parametric representation for each of
+    those objects.
+
+    Initialization
+    --------------
+
+    @param file_name    The file containing the catalog. [default: None, which will look for the
+                        COSMOS catalog in $PREFIX/share/galsim.  It will raise an exception if the
+                        catalog is not there telling you to run galsim_download_cosmos.]
+    @param use_real     Use realistic galaxies or parametric ones?  [default: True]
+    @param preload      Keyword that is only used for real galaxies, not parametric ones, to choose
+                        whether to preload the header information.  If `preload=True`, the bulk of  
+                        the I/O time is in the constructor.  If `preload=False`, there is
+                        approximately the same total I/O time (assuming you eventually use most of
+                        the image files referenced in the catalog), but it is spread over the
+                        various calls to getGal() and getPSF().  [default: False]
+    @param image_dir    Keyword that is only used for real galaxies, not parametric ones, to specify
+                        the directory of the image files.
+                        If a string containing no `/`, it is the relative path from the location of
+                        the catalog file to the directory containing the galaxy/PDF images.
+                        If a path (a string containing `/`), it is the full path to the directory
+                        containing the galaxy/PDF images. [default: None]
+    @param dir          The directory of catalog file. [default: None]
+    @param noise_dir    Keyword that is only used for real galaxies, not parametric ones.
+                        The directory of the noise files if different from the directory of the 
+                        image files.  [default: image_dir]
+    @param deep_sample  Modify fluxes and sizes of galaxies in order to roughly simulate an F814W<25
+                        sample? [default: False]
+    @param exclude_fail For catalogs of parametric galaxies, exclude those that have failures in the
+                        parametric fits?  [default: True]
+
+    Attributes
+    ----------
+
+    After construction, the following attributes are available:
+
+    nobjects     The number of objects in the catalog
+    obj_type     Either 'real' or 'parametric', depending on which way of representing galaxies was
+                 chosen.
+    deep         Either 'True' or 'False', depending on whether modifications of the object sizes
+                 and fluxes were used to simulate a deeper I<25 sample ('True') or not.
+
+    """
```

Comment:
I started working on this, and there are a few API changes I'd like to make.  Before doing so, I wanted to run them by you:
- Change the function name from `makeObj` to `makeGalaxy` since these are always galaxies.
- Have `makeGalaxy` take a `gal_type` parameter, rather than fix it at the time of `COSMOSCatalog` creation.
- The constructor would still take a `use_real` parameter to let people opt out of using any real galaxies, in which case the `gal_type` parameter for `makeGalaxy` defaults to 'parametric', since that's the only option.
- Change the name of the `pad_size` parameter to `noise_pad_size` to match what it is called in RealGalaxy.  Seems better to use the same name for the same concept.
- Don't multiply this number by sqrt(2).  I think I see the logic of why you did that, but I think it's better to just explain to people how they should calculate a good value for `noise_pad_size` rather than take what they give and then make it larger.
- Change the default `noise_pad_size` to 0.4 to accommodate the fact that 0.25 won't by multiplied by sqrt(2) anymore.  (Or should this be 0.35?  Or 0.5 if these sound to precise for the level of arbitrariness involved?)
- Change the name of `indices` to `index`.  Still allow it to be a list though.
- Have `makeObj` use the provided `rng` for noise padding as well as random index selection.
- Use the name 'COSMOSGalaxy' for this type of object in config.  There would be no corresponding python class.  It would just be the output of `COSMOSCatalog.makeGalaxy()`.

Feel free to push back on any of these if you don't like them.  

I'm planning to add this feature to demo11 for the galaxy selection, since the current galaxy choice is basically a rehash of demo10, except for the noise_pad_size option, so it would be fine to switch to COSMOSGalaxy there.  I was thinking of using 'RandomBinomial' to randomly select some fraction to be real and the rest parametric, since we don't demo that option anywhere yet.  Also it will be faster if we have some parametric galaxies than if they are all real.